### PR TITLE
v0.4.0

### PR DESCRIPTION
**Pull request by lebovic**
* Add Poetry, remove Twine

* Add CircleCI automatic deploy to PyPI (untested for prod PyPI)

Note: CircleCI will be failing because v0.4.0 already exists on test PyPI. That is to be expected, because I already bumped it to v0.4.0 when testing.

### CODE REVIEW

**Comment by bcai2** [comment]
I'm planning to merge the `bowtie2` branch in as part of this update. Are there any issues (technical, your preference, or otherwise) with trying to do this in the same merge/PR?

**Comment by bcai2** [comment]
This "review" is just going to be a stand-in for you checking over the bowtie2 code. Let me know if it looks good to merge.

File: toolchest_client/query.py [hunk 1]
```diff
@@ -134,8 +140,9 @@ def _send_initial_request(self, tool_name, tool_version,
         create_body = {
             "tool_name": tool_name,
             "tool_version": tool_version,
-            "database_name": None,
-            "database_version": None,
+            "custom_tool_args": tool_args,
```
Comment (lebovic) [inline toolchest_client/query.py:143]: 👌

**Review by lebovic — COMMENTED** [review]
LGTM! Let's start making the PR descriptions meaningful now that this is publicly exposed.

**Review by bcai2 — APPROVED** [review]
Everything looks good, then. I'll write the PR description when merging

**Pull request merged**